src/torrent.cpp

```diff
@@ -1138,6 +1138,20 @@ bool is_downloading_state(int const st)
 		if (m_peer_list) m_peer_list->clear();
 	}
 
+	void torrent::set_sequential_range(piece_index_t start_piece, piece_index_t end_piece)
+	{
+		m_sequential_download = true;
```

Comment:
I have changed the placement of this statement in the latter commits to set sequential downloading mode regardless if calling sequential range fails (eg. metadata hasn't been downloaded yet.) Earlier this statement was a few lines down, after the body of the has_picker predicate, so that the mode is set only if the call succeeds. I am open for your opinion on this.

Reply:
it seems reasonable. Just make sure the new behavior is documented